sdcm/utils/quota.py

```diff
@@ -0,0 +1,120 @@
+    node.remoter.sudo(conf_quota_cmd, verbose=True)
+    verify_quota_is_configured_for_scylla_user(node)
+    try:
+        yield quota_size
+    finally:
+        remove_quota_configuration_from_scylla_user_on_node(node)
+
+
+def verify_quota_is_configured_for_scylla_user(node):
+    """Verifies quota ia actually configured for scylla user, but doesn't verify enforcement in ON"""
+    get_scylla_user_quota_value = 'xfs_quota -x -c "report -h" /var/lib/scylla | grep scylla | tail -1 | awk {\'print $4\'}'
+    LOGGER.info("Verifying quota is configured for scylla user")
+    quota_value = node.remoter.sudo(get_scylla_user_quota_value).stdout.rstrip()
+    if quota_value == "0":
+        LOGGER.error("Failed to configure quota on user")
+        raise QuotaConfigurationFailure(
+            "Quota was not configured for scylla user. The value for hard quota is: {}".format(quota_value))
+    LOGGER.info("Quota was configured for scylla user with value: {}".format(quota_value))
+
+
+def remove_quota_configuration_from_scylla_user_on_node(node):
+    LOGGER.info('Remove quota configuration from user \'scylla\' by setting it to 0')
+    quota_remove_cmd = "xfs_quota -x -c 'limit bsoft=0 bhard=0 scylla' /var/lib/scylla"
+    node.remoter.sudo(quota_remove_cmd, verbose=True)
+
+
+def write_data_to_reach_end_of_quota(node, quota_size):
+    LOGGER.debug('Watching the log for the relevant message for quota...')
+    quota_exceeded_appearances = node.follow_system_log(
+        patterns=[r'Shutting down communications due to I\/O errors until operator intervention: Disk error:.*Disk quota exceeded'])
+    node.remoter.sudo('chsh -s /bin/bash scylla')  # Enable login for scylla user
+
+    def approach_end_of_quota():
+        if bool(list(quota_exceeded_appearances)):
+            return True
+        currently_used_space = get_currently_used_space_by_scylla_user(node)
+        occupy_space_size = int((quota_size - currently_used_space) * 90 / 100)
+        occupy_space_cmd = f"su - scylla -c 'fallocate -l {occupy_space_size}K /var/lib/scylla/occupy_90percent.{time.time()}'"
+        try:
+            LOGGER.debug('Cost 90% free space on /var/lib/scylla/ by {}'.format(occupy_space_cmd))
+            node.remoter.sudo(occupy_space_cmd, Verbose=True)
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.warning("We should have reached the expected I/O error and quota has exceeded\n"
+                           "Message: {}".format(str(exc)))
+        return bool(list(quota_exceeded_appearances))
+
+    wait_for(func=approach_end_of_quota,
+             timeout=300,
+             step=10,
+             text="Waiting for 'Disk error: std::system_error (error system:122, Disk quota exceeded)' "
+             "string to appear in the node log",
+             throw_exc=False
+             )
+    node.remoter.sudo('rm -rf /var/lib/scylla/occupy_90percent.*')  # Remove the sparse files
```

Comment:
where is the code that clears this file ?

Reply:
or in other word, we need to make sure we always execute this part, this should be in one finally clause of some context manager, to make sure we clear those